src/components/CodeOverview/index.tsx

```diff
@@ -83,10 +87,15 @@ export class CodeOverviewBase extends React.Component<Props, State> {
     this.setState({ overviewHeight: null });
   };
 
-  setOverviewHeight = () => {
+  setOverviewHeight = ({ _setState = this.setState.bind(this) } = {}) => {
```

Comment:
ugh. Is this needed for the test case? How about we don't do this and we spy on the instance's method instead?

```diff
diff --git a/src/components/CodeOverview/index.spec.tsx b/src/components/CodeOverview/index.spec.tsx
index 69cd3d1..81c5bf9 100644
--- a/src/components/CodeOverview/index.spec.tsx
+++ b/src/components/CodeOverview/index.spec.tsx
@@ -18,6 +18,7 @@ import {
   fakeVersion,
   shallowUntilTarget,
   simulateLinterProvider,
+  spyOn,
 } from '../../test-helpers';
 import styles from './styles.module.scss';
 
@@ -233,10 +234,10 @@ describe(__filename, () => {
 
     root.setState({ overviewHeight: clientHeight });
 
-    const _setState = jest.fn();
-    instance.setOverviewHeight({ _setState });
+    const setState = spyOn(instance, 'setState');
+    instance.setOverviewHeight();
 
-    expect(_setState).not.toHaveBeenCalled();
+    expect(setState).not.toHaveBeenCalled();
   });
 
   it('can reset the overview height', () => {
diff --git a/src/components/CodeOverview/index.tsx b/src/components/CodeOverview/index.tsx
index 6ae1af9..349f2a3 100644
--- a/src/components/CodeOverview/index.tsx
+++ b/src/components/CodeOverview/index.tsx
@@ -87,14 +87,14 @@ export class CodeOverviewBase extends React.Component<Props, State> {
     this.setState({ overviewHeight: null });
   };
 
-  setOverviewHeight = ({ _setState = this.setState.bind(this) } = {}) => {
+  setOverviewHeight = () => {
     const { overviewHeight } = this.state;
     const ref = this.overviewRef;
 
     if (ref && ref.current) {
       const refHeight = ref.current.clientHeight;
       if (overviewHeight !== refHeight) {
-        _setState({ overviewHeight: refHeight });
+        this.setState({ overviewHeight: refHeight });
       }
     }
   };
```

Reply:
great idea, thanks! I thought this wasn't possible for React components but maybe I'm thinking of a sinon limitation or maybe it's fixed now.